pages/genre/genre.go

```diff
@@ -42,3 +45,22 @@ func containsLowerCase(array []string, search string) bool {
 
 	return false
 }
+
+func averageGenreScore(user *arn.User, animes []*arn.Anime) string {
+	if user == nil {
+		return ""
+	}
+
+	counter := 0.0
+	scores := 0.0
+
+	for _, anime := range animes {
+
+		if user.AnimeList().Contains(anime.ID) {
+			scores = scores + user.AnimeList().Find(anime.ID).Rating.Overall
```

Comment:
1.) Use cached `animeList` here.
2.) Using `Find` after `Contains` is `O(2n)` performance behavior.
You can optimize it to `O(n)` by only using `Find` and checking for a `nil` item instead of using `Contains`.

Reply:
@Amatrelan You might also want to take only the AnimeListItems that have are actually rated by the user, you can get it like so: `item.Rating.IsNotRated()`